### PR TITLE
Fix unit tests for ann_secret PR

### PR DESCRIPTION
**Pull request by ggriffiths**
Fixed failing unit tests. 

I haven't added any new ones - let me know if you think there are more cases that need to be added.

The tests check for valid, empty, and invalid secrets.

### CODE REVIEW

File: pkg/controller/snapshot_controller.go [hunk 1]
```diff
@@ -694,11 +694,11 @@ func (ctrl *csiSnapshotController) createSnapshotOperation(snapshot *crdv1.Volum
 	}
 
 	// Set AnnDeletionSecretRefName and AnnDeletionSecretRefNamespace
-	if !metav1.HasAnnotation(snapshotContent.ObjectMeta, AnnDeletionSecretRefName) {
+	if !metav1.HasAnnotation(snapshotContent.ObjectMeta, AnnDeletionSecretRefName) && snapshotterSecretRef != nil {
```
Comment (ggriffiths) [inline pkg/controller/snapshot_controller.go:697]: This is the only non-test code I added.

We were getting a nil pointer here when trying to access a nil `snapshotterSecretRef`

**Comment by xing-yang** [comment]
This looks good!  Let's wait until this PR (https://github.com/kubernetes-csi/external-snapshotter/pull/139) is merged, then I'll rebase mine.  Then we can try to get the secrets PRs merged.

**Comment by ggriffiths** [comment]
Sounds good :+1:

**Comment by xing-yang** [comment]
Actually I think we should get this merged now so that unit tests of that secret PR can pass:).

**Pull request merged**